config/localTemplate.go

```diff
@@ -237,10 +237,6 @@ type Local struct {
 	// the max size the sync server would return
 	TxSyncServeResponseSize int `version[3]:"1000000"`
 
-	// IsIndexerActive indicates whether to activate the indexer for fast retrieval of transactions
-	// Note -- Indexer cannot operate on non Archival nodes
-	IsIndexerActive bool `version[3]:"false"`
```

Comment:
Need to leave this in or else we'd config file parsing would fail. Even though most configs will have it set to false already, I'm pretty sure unknown fields are illegal.

Reply:
I thought that too, but check out the config test - it's loading config versions prior to v28 that have the property without error. I then looked at the code, it's looping over the properties in the local struct (rather than what is in the config file on disk).

Reply:
Woah. I wonder if its always been that way. Thanks for the correction!

Reply:
https://github.com/algorand/go-algorand/blob/master/config/config_test.go#L292

Reply:
> // To unmarshal JSON into a struct, Unmarshal matches incoming object
// keys to the keys used by Marshal (either the struct field name or its tag),
// preferring an exact match but also accepting a case-insensitive match. By
// default, object keys which don't have a corresponding struct field are
// ignored (see Decoder.DisallowUnknownFields for an alternative).

indeed, what a surprize, I thought it fails like msgp

Reply:
I checked IsIndexerActive=true also deserialized without problems and matches the documentation.

Not sure tho if silently removal config values is a good practice? Maybe there should be some kind of deprecated fields and warning in node.log?